### DISCUSSION TITLE
It seems that `excl_paths` is not working

### DISCUSSION THREAD

**Discussion by zoziha** [fortran-lang/fortls#218] [General]
### Description

I mainly use `Windows` + `VS Code` + `Modern Fortran` + `fortls` + `fprettify` + `ford`.

Recently, since some time, `fortls` does not seem to handle correctly the `excl_paths` which I set in `settings.json`.

**I have a [replication example here](https://github.com/zoziha/fortls-demo), when I click on the variable `a` in `app/main.f90` and go to the definition, it will jump to `doc/src/mod.f90`(`ford` `output_dir: doc`) instead of `src/mod.f90`.**

**Here is my `settings.json`:**
```json
{
    "fortran.fortls.excludeDirectories": [
        "doc/**"
    ],
}
```

I had tried to mention this as an issue, but it seems like the requirements for writing an issue are more complicated.
Does the `excl_paths` failure actually exist, or is it a problem with my settings? Thanks!

**Comment by gnikit** [comment]
Apologies for taking some time to get back to you @zoziha. 
You have stumbled onto a bug in fortls/Modern Fortran vscode. What is happening as far as I can tell is that when you open `app/main.f90` fortls parses your workspace excluding `docs/`.
However, the moment you open a file from within the excluded directory `docs/` then fortls parses that that file too and conflicting definitions get overwritten.

On a high level, what should be happening is this situation is discard the states of the excluded files when they are closed.
Practically, how that would be done is probably a more complicated matter.

Other than closing the source files from the excluded directories and restarting the language server (via the Command Palette) I don't have a very good temporary fix for this. The matter definitely needs further investigation.

**Reply by zoziha** [reply]
Thank you for your reply, @gnikit !

I checked and it seems that **when I open `app/main.f90`, the source files for both `doc/src` and `src` are parsed**:

In my reproduced example, there are 5 source files in total:
- app/main.f90;
- doc/src/main.f90;
- doc/src/mod.f90;
- src/mod.f90;
- test/check.f90.

<details>
<summary>3 source files (+ 2 source files in doc/src)</summary>

![image](https://user-images.githubusercontent.com/32035096/200211164-d94cbe8c-ad2b-456d-b01e-5a186755634e.png)

</details>

When I open `app/main.f90`, the status bar tells me that `fortls` is parsing 5 source files instead of 3.

<details>
<summary>Status bar information</summary>

![vscode](https://user-images.githubusercontent.com/32035096/200211113-1d22d349-72ee-4213-aaee-c0aabf78a4df.gif)

</details>
